core/kazoo_attachments/src/kz_att_s3.erl

```diff
@@ -20,18 +20,15 @@
 
 -spec put_attachment(kz_data:connection(), ne_binary(), ne_binary(), ne_binary(), ne_binary(), kz_data:options()) -> any().
 put_attachment(Params, DbName, DocId, AName, Contents, _Options) ->
-    #{bucket := Bucket
-     ,key := Key
-     ,secret := Secret
-     ,path := Path
-     } = Params,
+    {Bucket, Key, Secret, Path, Host} = get_map_values(Params),
     FilePath = kz_util:to_list(list_to_binary([Path, "/", DbName, "/", DocId, "_", AName])),
-    Config = kz_aws_s3:new(kz_util:to_list(Key), kz_util:to_list(Secret)),
+    Config = kz_aws_s3:new(kz_util:to_list(Key), kz_util:to_list(Secret), kz_util:to_list(Host)),
     case kz_aws_s3:put_object(kz_util:to_list(Bucket), FilePath, Contents, Config) of
         {'ok', Props} ->
             Metadata = [ convert_kv(KV) || KV <- Props, filter_kv(KV)],
             S3Key = base64:encode(term_to_binary({Key, Secret, Bucket, Path})),
             {'ok', [{'attachment', [{<<"S3">>, S3Key}
+                                   ,{<<"Host">>, Host}
```

Comment:
i am handling the filepath in another fun so that an empty path doesn't cause issue here
https://github.com/tsromana/kazoo/blob/838ffc97de1517cc6dbacbecd6fbcd075ed24ff2/core/kazoo_attachments/src/kz_att_s3.erl#L24